### ISSUE TITLE
Devices can not ping each other using their IPv4 addresses

### ISSUE THREAD

**Issue by rogerpueyo** [bmx-routing/bmx6#12]
I'm runnnig a few nodes with different versions of qMp which, in turn, run diverse versions of BMX6. Starting with version https://github.com/bmx-routing/bmx6/commit/5dc6678cf9c2887ca5e32c8d7527c5f660ddb7e9 , devices can not ping each other using their announced IPv4 address(es), but can do it using their IPv6 ones.

If I replace the /usr/sbin/bmx6 file from a non-functional node with that from a working one, which has version https://github.com/bmx-routing/bmx6/commit/4016a1980d900309771e432d1f7c741d6c48d477 or older, IPv4 pings work fine.

**Comment by axn** [comment]
Maybe the 5dc6678 does not work with older kernels. I think I tested with a recent lede one, but maybe not extensively enough. Which kernel are you running? And could you update to a newer version?

**Comment by p4u** [comment]
It might make me understand the crazy behavior I've observed in the last libremesh builds... I've to double check it but I think I experienced the same problem.

**Comment by rogerpueyo** [comment]
@axn,

I'm running kernel 4.4.92, the one shipped with LEDE 17.01.4, but I am experiencing the same with kernel 4.9 from OpenWrt trunk.

**Comment by p4u** [comment]
Yes, going back to previous version of bmx6 ipv4 works again.

I'm using kernel 4.4.92 and LEDE 17.01.4

**Comment by rogerpueyo** [comment]
Please check the following messages in the [qmp-users mailing list](https://mail.dabax.net/cgi-bin/mailman/listinfo/qmp-users) for a report a user made:

https://mail.dabax.net/pipermail/qmp-users/2018-January/001088.html
https://mail.dabax.net/pipermail/qmp-users/2018-January/001089.html
https://mail.dabax.net/pipermail/qmp-users/2018-January/001094.html
https://mail.dabax.net/pipermail/qmp-users/2018-January/001095.html

**Comment by nicoechaniz** [comment]
reverting to commit #8ba44134d9c9067d9db356a5b1a82d71f133ea6d  routing is working fine

**Comment by axn** [comment]
I eventually tracked down the problem to a particular kernel patch of openwrt which eventually breaks the fall-back capabilities of the ip4-in-ipv6 tunnel feature. Till this is fixed I also reverted the critical commit in openwrt-routing-packes master branch 21a6454d7226eb65863822e62745f7d49e6d9b41 (in lede-17.04 branch it got reverted already earlier via b82d7ccd0abe042359c41229090c2ce5d4276950). 
Trying to get this handled upstream in openwrt: https://lists.openwrt.org/pipermail/openwrt-devel/2018-February/043510.html

Thanks for helping and sorry for the trouble.
Closing this issue for now.
